collector/node_collector.go

```diff
@@ -114,43 +123,49 @@ func (a *Agent) getIP() (string, error) {
 		return "", err
 	}
 
-	log.Debugf("getIP() returned successfully, got IP %s", ip)
+	nodeColLog.Debugf("getIP() returned successfully, got IP %s", ip)
 	return ip, nil
 }
 
-// pollAgent queries the DC/OS agent for metrics and returns.
-func (a *Agent) pollAgent() metricsMeta {
+// pollHost queries the DC/OS hsot for metrics and returns.
+func (h *DCOSHost) pollHost() metricsMeta {
 	metrics := metricsMeta{}
 	now := time.Now().UTC()
-	log.Infof("Polling the Mesos agent at %s:%d. Started at %s", a.AgentIP, a.Port, now.String())
+	nodeColLog.Infof("Polling the DC/OS host at %s:%d. Started at %s", h.IPAddress, h.Port, now.String())
 
-	// always fetch/emit agent state first: downstream will use it for tagging metrics
-	log.Debugf("Fetching state from agent %s:%d", a.AgentIP, a.Port)
-	agentState, err := a.getAgentState()
-	if err != nil {
-		log.Errorf("Failed to get agent state from %s:%d. Error: %s", a.AgentIP, a.Port, err)
-		return metrics
-	}
+	if h.DCOSRole == "agent" {
```

Comment:
this should probably be a constant. https://github.com/dcos/dcos-go/pull/15 will have these constants exported https://github.com/dcos/dcos-go/blob/29e9e7508543aedf36a8bbccaffdf87d1fbe0f20/dcos/constants.go you can reuse one PR 15 land